### PR TITLE
Copy edit nddata

### PR DESCRIPTION
**Pull request by lglattly**
Copy edits to the nddata sub-package!

### CODE REVIEW

**Comment by pllim** [comment]
There is a formatting warning that needs to be fixed, FYI:
```
Warning, treated as error:
/home/circleci/project/docs/nddata/index.rst:367:Title level inconsistent:
```

**Comment by adrn** [comment]
Ah, yea - it looks like all of the "Examples" headings were added as "=====", but I guess the heading level should depend on where the examples are relative to the enclosing heading, right?

**Comment by lglattly** [comment]
Hmm yeah that is an issue I was running into in the IO subpackage as well with trying to flag all of the examples for the Example Gallery. I had thought there was a convention in Astropy that any example heading needs the "=====" level heading, but maybe that is incorrect?

@jonathansick and @kelle do you remember if we discussed that the examples headings had to have the "=====" level heading or can they have "-------" if that is what level the enclosing heading has?

**Comment by jonathansick** [comment]
The `example` directive won't place any constraint on heading levels. So you can use whatever level makes sense given the hierarchy (as @adrn says).

**Comment by lglattly** [comment]
Ok great thanks! will fix

**Comment by lglattly** [comment]
Ok hopefully that fixes it. Also @adrn can you double check that the example titles I put in the notes make sense? I did my best to come up with them from context but this is definitely where I'll need auditing from the experts!

File: docs/nddata/index.rst [hunk 1]
```diff
@@ -185,22 +199,31 @@ WCS, sliced appropriately also::
...
+This example creates an uncertainty that is Poisson error, stored as a
+variance::
 
     >>> from astropy.nddata import VarianceUncertainty
     >>> poisson_noise = np.ma.sqrt(np.ma.abs(ccd.data))
     >>> ccd.uncertainty = VarianceUncertainty(poisson_noise ** 2)
 
-As a convenience, the uncertainty can also be set with a numpy array. In that
-case, the uncertainty is assumed to be the standard deviation::
+As a convenience, the uncertainty can also be set with a ```numpy`` array. In
```
Comment (MSeifert04) [inline docs/nddata/index.rst]: One backtick too much.

Comment (lglattly) [reply docs/nddata/index.rst]: Good catch!

**Comment by lglattly** [comment]
Thanks @MSeifert04! And yes, the style guide convention I've been following for Astropy and NumPy is to use the proper noun when referring to The Project, and code formatting when referring to the core package. I had assumed that a numpy array refers more to code than the overall project, but do you think that's correct?

File: docs/nddata/index.rst (hunk 1 above)
Comment (adrn) [reply docs/nddata/index.rst]:
```suggestion
As a convenience, the uncertainty can also be set with a ``numpy`` array. In
```

File: docs/nddata/subclassing.rst [hunk 2]
```diff
@@ -215,22 +232,35 @@ be anything except ``None`` or ``"first_found"``::
   To use these internal `_arithmetic_*` methods there are some restrictions on
   the attributes when calling the operation:
 
-  - ``mask``: ``handle_mask`` must not be ``None``, ``"ff"`` or ``"first_found"``.
+  - ``mask``: ``handle_mask`` must not be ``None``, ``"ff"``, or
+  ``"first_found"``.
```
Comment (adrn) [inline docs/nddata/subclassing.rst]:
```suggestion
  ``  "first_found"``.
```

File: docs/nddata/mixins/ndarithmetic.rst [hunk 3]
```diff
@@ -63,39 +63,39 @@ or to wrap the result of an arithmetic operation between two Quantities::
...
-----------
+Data and Unit
+-------------
 
 For ``data`` and ``unit`` there are no parameters. Every arithmetic
 operation lets the `astropy.units.Quantity`-framework evaluate the result
 or fail and abort the operation.
 
-Adding two NDData objects with the same unit works::
+Adding two `NDData` objects with the same unit works::
```
Comment (adrn) [inline docs/nddata/mixins/ndarithmetic.rst]:
```suggestion
Adding two `~astropy.nddata.NDData` objects with the same unit works::
```

File: docs/nddata/mixins/ndarithmetic.rst [hunk 4]
```diff
@@ -106,7 +106,7 @@ Adding two NDData objects with the same unit works::
     >>> ndd.unit
     Unit("m")
 
-Adding two NDData objects with compatible units also works::
+Adding two `NDData` objects with compatible units also works::
```
Comment (adrn) [inline docs/nddata/mixins/ndarithmetic.rst]:
```suggestion
Adding two `~astropy.nddata.NDData` objects with compatible units also works::
```

**Comment by bsipocz** [comment]
Yes, 32bit failures are fixed on master, they should go away if you rebase (also abit of squashing  of the single suggestions into one commit wouldn't hurt, but it's not a strong preference)

**Pull request merged**